packages/bun-usockets/src/loop.c

```diff
@@ -391,7 +391,27 @@ void us_internal_dispatch_ready_poll(struct us_poll_t *p, int error, int eof, in
                       const int recv_flags = MSG_DONTWAIT | MSG_NOSIGNAL;
                     #endif
 
-                    int length = bsd_recv(us_poll_fd(&s->p), loop->data.recv_buf + LIBUS_RECV_BUFFER_PADDING, LIBUS_RECV_BUFFER_LENGTH, recv_flags);
+                    int length;
+                    if(s->context->is_ipc) {
```

Comment:
can you put this on the socket type instead so we avoid this branch?

Reply:
the branch is just before a syscall, is it important to avoid it?